map/framework.cpp

```diff
@@ -2178,6 +2178,14 @@ place_page::Info Framework::BuildPlacePageInfo(place_page::BuildInfo const & bui
     // Selection circle should match feature
     FillFeatureInfo(selectedFeature, outInfo);
 
+    if (!outInfo.IsBookmark() && !isBuildingSelected)
+    {
+      // Search for a bookmark at POI position instead of tap position
+      auto mark = FindBookMarkInPosition(outInfo.GetMercator());
```

Comment:
Searching _all_ bookmarks in _all_ lists twice on each tap looks like an overhead. @vng do you have any better ideas on the implementation? The tap's radius doesn't find a bookmark near it but selects a POI at a distance with a large text which was tapped.